### PR TITLE
Fix capes being scaled wrong and invisible skins

### PR DESCRIPTION
**Pull request by rtm516**
Fixes:
image

### CODE REVIEW

File: connector/src/main/java/org/geysermc/connector/skin/SkinProvider.java [hunk 1]
```diff
@@ -412,15 +412,35 @@ private static Skin supplyEars(Skin existingSkin, String earsUrl) {
...
+            BufferedImage newImage;
             if (image.getWidth() > 64) {
-                image = scale(image, 64, 32);
+                newImage = new BufferedImage(128, 64, BufferedImage.TYPE_INT_ARGB);
+                Graphics g = newImage.createGraphics();
+                g.drawImage(image, 0, 0, image.getWidth(), image.getHeight(), null);
+                g.dispose();
+                newImage = scale(newImage, 64, 32);
+            } else {
+                newImage = new BufferedImage(64, 32, BufferedImage.TYPE_INT_ARGB);
```
Comment (Camotoy) [inline connector/src/main/java/org/geysermc/connector/skin/SkinProvider.java:424]: Is this also for removing alpha? If so, add a comment?

File: connector/src/main/java/org/geysermc/connector/skin/SkinProvider.java [hunk 2]
```diff
@@ -412,15 +412,35 @@ private static Skin supplyEars(Skin existingSkin, String earsUrl) {
...
             if (image.getWidth() > 128) {
                 image = scale(image, 128, image.getHeight() / (image.getWidth() / 128));
             }
+
+            // Remove alpha from skins to replicate how java displays
+            BufferedImage newImage = new BufferedImage(image.getWidth(), image.getHeight(), BufferedImage.TYPE_INT_RGB);
+            Graphics g = newImage.createGraphics();
+            g.drawImage(image, 0, 0, image.getWidth(), image.getHeight(), null);
+            g.dispose();
+            image = newImage;
```
Comment (Camotoy) [inline connector/src/main/java/org/geysermc/connector/skin/SkinProvider.java:443]: Flush the old image?

File: connector/src/main/java/org/geysermc/connector/skin/SkinProvider.java (hunk 1 above)
Comment (rtm516) [reply connector/src/main/java/org/geysermc/connector/skin/SkinProvider.java:424]: It shouldn't nor does it need to.

Comment (Camotoy) [reply connector/src/main/java/org/geysermc/connector/skin/SkinProvider.java:424]: I'm not sure why we need this else statement, then. I tested a normal-scale cape before this PR and it was fine.

File: connector/src/main/java/org/geysermc/connector/skin/SkinProvider.java [hunk 3]
```diff
@@ -412,15 +412,38 @@ private static Skin supplyEars(Skin existingSkin, String earsUrl) {
...
             }
+
+            // Remove alpha from skins to replicate how java displays
+            BufferedImage newImage = new BufferedImage(image.getWidth(), image.getHeight(), BufferedImage.TYPE_INT_RGB);
+            Graphics g = newImage.createGraphics();
+            g.drawImage(image, 0, 0, image.getWidth(), image.getHeight(), null);
+            g.dispose();
+
+            image.flush();
+            image = newImage;
```
[the remark this answers is not in the document]
Comment (rtm516) [reply connector/src/main/java/org/geysermc/connector/skin/SkinProvider.java:446]: Can you upload your skin?

[the remark this answers is not in the document]
Comment (Camotoy) [reply connector/src/main/java/org/geysermc/connector/skin/SkinProvider.java:446]: This is a result of this PR also setting the alpha channel on the helmet layer. I can confirm it's not just an issue with one skin.

[the remark this answers is not in the document]
Comment (HookWoods) [reply connector/src/main/java/org/geysermc/connector/skin/SkinProvider.java:446]: https://namemc.com/skin/e4b881911fca8d6a
Here you go

**Review by Camotoy — APPROVED** [review]

**Pull request merged**